src/compiler.ts

```diff
@@ -6477,7 +6503,11 @@ export class Compiler extends DiagnosticEmitter {
       let declaration = originalParameterDeclarations[minArguments + i];
       let initializer = declaration.initializer;
       let initExpr: ExpressionRef;
-      if (initializer) {
+      if (declaration.parameterKind === ParameterKind.Rest) {
+        const arrExpr = new ArrayLiteralExpression([], declaration.range.atEnd);       
+        initExpr = this.compileArrayLiteral(arrExpr, type, Constraints.ConvExplicit);
```

Comment:
Is it possible to use `compileStaticArrayLiteral` here?

Reply:
Good question.  I will attempt it and let you know! 👍

Reply:
Reading `compileArrayLiteral`'s code, it looks like the user can opt into `StaticArray` by declaring `...args: StaticArray<i32>`. @mattjohnsonpint maybe you should make a test for that...

Anyhow, the user might desire the admittedly strange case where the function might mutate and play around with the `args` array (or perhaps more commonly, use it to initialize a class field?).

Reply:
Nope.  It doesn't work.  It would be the equivalent of doing this:

```ts
function foo<T>(args: T[] = new StaticArray<T>())
```

and a static array can't be assigned to a regular array.